src/Native/Runtime/amd64/InteropThunksHelpers.S

```diff
@@ -4,19 +4,43 @@
 
 .intel_syntax noprefix
 #include <unixasmmacros.inc>
-
-//TODO: Implement these stub method for non-windows
+#include <AsmOffsets.inc>         // generated by the build from AsmOffsets.cpp
+#define POINTER_SIZE 8
 
 LEAF_ENTRY RhpCommonStub, _TEXT
-    int 3
+
+    PUSH_ARGUMENT_REGISTERS
+
+    // +8 for alignment
+    alloc_stack    (SIZEOF_MAX_FP_ARG_SPILL + 8)
```

Comment:
The constants in the asmconstants.h need to be updated according to the calling convention. On Unix amd64, the arguments can be passed in xmm0..xmm7 while the constant is defined to cover just xmm0..3